stateproof/worker.go

```diff
@@ -118,10 +118,11 @@ func (spw *Worker) Start() {
 	go spw.builder(latest)
 }
 
```

Comment:
it looks like on 'Start' we call `spw.net.RegisterHandle`. Should we remove the registration?

Reply:
+ should we also add `spw.wg.Done()` to `signer` and `builder` ?

@algorandskiy , @algonautshant thoughts?

Reply:
> it looks like on 'Start' we call spw.net.RegisterHandler. Should we remove the registration?

Since we reload the ledger and restart trackers, in `Stop` need to unregister, and register in `Start` again.

> Is that okay that we don't have spw.wg.Done()

If we start goroutines in `Start`, need to finish them in `Stop` by cancelling the context (and context creation moves to Start() as well).

Reply:
See my comments here about deregistering, and why it is important to do so:
https://github.com/algorand/go-algorand/pull/4803#pullrequestreview-1290160638

Reply:
Apparently, the network interface doesn't allow unregistering the network handler (The only option is to remove all handlers). it is not optimal, but I don't want to change the design at the moment.

 I've made sure that we unregister the network handler before  calling `Stop`